src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImplAsync.cs

```diff
@@ -975,7 +975,7 @@ private async Task InitStreamInputAsync(Uri baseUri, string baseUriStr, Stream s
             _ps.bytePos = 0;
             while (_ps.bytesUsed < 4 && _ps.bytes.Length - _ps.bytesUsed > 0)
             {
-                int read = await stream.ReadAsync(_ps.bytes, _ps.bytesUsed, _ps.bytes.Length - _ps.bytesUsed).ConfigureAwait(false);
+                int read = await stream.ReadAsync(_ps.bytes.AsMemory(_ps.bytesUsed, _ps.bytes.Length - _ps.bytesUsed)).ConfigureAwait(false);
```

Comment:
We should ideally update the fixer to detect this. Namely if the count is array.Length - offset, use the single-parameter AsMemory instead.

Reply:
I'll revert the two changes pointed out by @GrabYourPitchforks , update the fixer and then come back to runtime to make sure the fixer detects them for me. Sounds good?

Reply:
Thinking about this more, I'll just apply it now, and when it comes time to test it in runtime, I'll temporarily revert this change locally to see if it would've worked, but that way I don't have to submit another PR.

Reply:
@stephentoub I think we should also have a _separate_ Roslyn analyzer that detects when `buffer.AsMemory(offset, buffer.Length - offset)` is being used, and suggest using `buffer.AsMemory(offset)`, unless such analyzer already exists. What do you think? I can open the issue if you're ok with that.

Reply:
I opened [an issue ](https://github.com/dotnet/runtime/issues/35981) suggesting the creation of an analyzer/fixer for `AsMemory`. Let me know what you think.

Reply:
I think it's reasonable in addition.  We should ensure this fixer generates the best code we can muster, but then being able to address other inefficiencies is useful, too.

Reply:
Awesome. I also opened [this issue](https://github.com/dotnet/roslyn-analyzers/issues/3612) to track the addition of this improvement to the existing analyzer, plus another improvement I just thought of: If offset=0 and count=buffer.Length, then `AsMemory()` should have no parameters.